code/modules/organs/internal/_internal.dm

```diff
@@ -135,22 +149,50 @@
 	. = "[.][name]"
 
 /obj/item/organ/internal/Process()
+	SHOULD_CALL_PARENT(TRUE)
 	..()
-	handle_regeneration()
-
-/obj/item/organ/internal/proc/handle_regeneration()
-	if(!damage || BP_IS_PROSTHETIC(src) || !owner || GET_CHEMICAL_EFFECT(owner, CE_TOXIN) || owner.is_asystole())
-		return
-	if(damage < 0.1*max_damage)
-		heal_damage(0.1)
+	if(owner && damage && !(status & ORGAN_DEAD))
+		handle_damage_effects()
+
+/obj/item/organ/internal/proc/handle_damage_effects()
+	SHOULD_CALL_PARENT(TRUE)
+	if(organ_can_heal())
+		// Some logic to prevent us healing straight past the threshold just by virtue of the heal amount being larger than the remaining damage in the current threshold.
+		var/min_heal_val = (min_regeneration_cutoff_threshold && !past_damage_threshold(min_regeneration_cutoff_threshold)) ? 0 : (get_current_damage_threshold() * damage_threshold_value)
```

Comment:
Can we make this more readable? Break out the `&&`, avoid the unnecessary negation in the ternary, etc.

Reply:
`!A ? B : C` is equivalent to `A ? C : B` which is one less logical operation to parse when reading. But you can write it however you prefer, I just had to read this line like 10 times to understand it.